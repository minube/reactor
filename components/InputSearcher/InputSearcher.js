import {
  bool, func, shape, string,
} from 'prop-types';
import React, { PureComponent } from 'react';
import { View, FlatList } from 'react-native';

import Text from '../Text';
import Input from '../Input';
import Touchable from '../Touchable';
import styles from './InputSearcher.style';

const TIMEOUT = 500;

class Searcher extends PureComponent {
  static propTypes = {
    color: string,
    data: shape({}),
    disabled: bool,
    hint: string,
    label: string,
    onChange: func,
    onClickItem: func,
    value: string,
  };

  static defaultProps = {
    color: undefined,
    data: undefined,
    disabled: false,
    hint: undefined,
    label: undefined,
    onChange: undefined,
    onClickItem: undefined,
    value: undefined,
  };

  constructor(props) {
    super(props);
    const { value } = props;

    this.searcherTimeout = undefined;
    this.state = { inputValue: value, data: [] };
  }

  _onChange = (value) => {
    const { onChange } = this.props;

    this.setState({ inputValue: value });

    if (value.length > 2) {
      if (this.searcherTimeout) clearTimeout(this.searcherTimeout);

      this.searcherTimeout = setTimeout(async () => {
        const response = await onChange(value);
        this.setState({ data: response });
      }, TIMEOUT);
    } else {
      this.setState({ data: [] });
    }
  }

  _onClickItem = item => this.setState({ data: {}, inputValue: item.name });

  render() {
    const {
      _onChange, _onClickItem,
      state: {
        inputValue, data,
      },
      props: { value, ...inherit },
    } = this;

    return (
      <View>
        <Input
          {...inherit}
          icon="search"
          onChange={values => _onChange(values)}
          placeholder="Look for ..."
          value={inputValue}
        />
        <FlatList
          data={data}
          style={styles.list}
          renderItem={({ item }) => (
            <Touchable onPress={() => _onClickItem(item)}>
              <Text style={styles.item} numberOfLines={1}>{item.name}</Text>
            </Touchable>
          )}
        />
      </View>
    );
  }
}

export default Searcher;
