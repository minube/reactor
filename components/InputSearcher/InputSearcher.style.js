import { StyleSheet } from 'react-native';

import { THEME } from '../../common';

const {
  BORDER_RADIUS, COLOR, UNIT, SPACE,
} = THEME;

const HEIGHT = UNIT * 4.4;

export default StyleSheet.create({
  list: {
    backgroundColor: COLOR.WHITE,
    borderRadius: BORDER_RADIUS,
    paddingHorizontal: SPACE.S,
    position: 'absolute',
    top: HEIGHT,
    width: '100%',
  },

  item: {
    paddingVertical: SPACE.XXS,
  },
});
