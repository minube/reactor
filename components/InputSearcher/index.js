export { default } from './InputSearcher'; // eslint-disable-line
