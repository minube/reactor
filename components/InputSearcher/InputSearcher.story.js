import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import {
  text, boolean, color, object,
} from '@storybook/addon-knobs/react';
import InputSearcher from './InputSearcher';

const DATA = [
  {name: 'Madrid', id: 23},
  {name: 'Madrid Snowzone', id: 12},
  {name: 'Comunidad Madrid', id: 65},
  {name: 'San Lorenzo del escorial', id: 76},
  {name: 'Alcalá de Henares', id: 132},
  {name: 'Aranjuez', id: 73},
];

const STYLE = { padding: 10, width: 256 };

_onChange = () => DATA;

storiesOf('✅ InputSearcher', module)
  .add('default', () => (
    <InputSearcher />
  ))
  .add('color', () => (
    <InputSearcher color="green" />
  ))
  .add('dataSource', () => (
    <InputSearcher dataSource={DATA} />
  ))
  .add('disabled', () => (
    <InputSearcher disabled />
  ))
  .add('hint', () => (
    <InputSearcher hint="Optional field" />
  ))
  .add('label', () => (
    <InputSearcher label="InputSearcher" />
  ))
  .add('placeholder', () => (
    <InputSearcher placeholder={"Looking for..."} />
  ))
  .add('value', () => (
    <InputSearcher value="Madrid" />
  ))
  .add('onChange', () => (
    <InputSearcher onChange={_onChange} />
  ))
  .add('🏀 Playground', () => (
    <InputSearcher
      color={color('color', undefined)}
      hint={text('hint', 'This is a hint')}
      disabled={boolean('disabled', false)}
      label={text('label', 'This is a label')}
      value={text('value', 'Madrid')}
      onChange={action('InputSearcher.onChange()')}
      style={object('style', STYLE)}
    />
  ));
