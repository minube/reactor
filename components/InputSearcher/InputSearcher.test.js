import React from 'react';
import renderer from 'react-test-renderer';
import InputSearcher from './InputSearcher';

const VALUE = 'Madrid';
const FUNC = () => {};
const STYLE = { backgroundColor: 'rgba(0,255,0,0.25)', padding: 10 };

describe('<InputSearcher>', () => {
  it('renders', () => {
    const tree = renderer.create(<InputSearcher />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {disabled}', () => {
    const tree = renderer.create(<InputSearcher disabled />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {hint}', () => {
    const tree = renderer.create(<InputSearcher hint="Hello World" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {label}', () => {
    const tree = renderer.create(<InputSearcher label="Hello World" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {value}', () => {
    const tree = renderer.create(<InputSearcher value={VALUE} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {onChange}', () => {
    const tree = renderer.create(<InputSearcher onChange={FUNC} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {style}', () => {
    const tree = renderer.create(<InputSearcher style={STYLE} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
