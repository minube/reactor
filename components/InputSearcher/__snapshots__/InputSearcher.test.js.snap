// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputSearcher> renders 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "marginBottom": 20,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "#cdd2d4",
            "borderRadius": 2.5,
            "borderWidth": 1,
            "display": "flex",
            "flexDirection": "row",
            "paddingHorizontal": 8,
          },
          false,
          false,
          false,
          undefined,
        ]
      }
    >
      <View
        pointerEvents="none"
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingHorizontal": 4,
          }
        }
      >
        <Image
          resizeMode="contain"
          source={
            Object {
              "testUri": "../../../components/Icon/assets/search.png",
            }
          }
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "height": 20,
                "width": 20,
              },
              Object {
                "alignSelf": "flex-start",
                "height": 18,
                "marginRight": 4,
                "width": 18,
              },
              undefined,
            ]
          }
        />
      </View>
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        blurOnSubmit={true}
        editable={true}
        keyboard="default"
        keyboardType="default"
        multiline={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder="Look for ..."
        placeholderTextColor="#707070"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
              "color": "#4a4a4a",
              "flex": 1,
              "fontFamily": undefined,
              "fontSize": 16,
              "fontWeight": "400",
              "minHeight": 42,
              "paddingHorizontal": 4,
              "paddingVertical": 7.5,
              "width": "100%",
            },
            false,
            undefined,
            undefined,
            undefined,
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "backgroundColor": "#ffffff",
        "borderRadius": 3,
        "paddingHorizontal": 12,
        "position": "absolute",
        "top": 44,
        "width": "100%",
      }
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;

exports[`<InputSearcher> when {disabled} 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "marginBottom": 20,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "#cdd2d4",
            "borderRadius": 2.5,
            "borderWidth": 1,
            "display": "flex",
            "flexDirection": "row",
            "paddingHorizontal": 8,
          },
          Object {
            "backgroundColor": "#f4f4f4",
          },
          false,
          false,
          false,
        ]
      }
    >
      <View
        pointerEvents="none"
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingHorizontal": 4,
          }
        }
      >
        <Image
          resizeMode="contain"
          source={
            Object {
              "testUri": "../../../components/Icon/assets/search.png",
            }
          }
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "height": 20,
                "width": 20,
              },
              Object {
                "alignSelf": "flex-start",
                "height": 18,
                "marginRight": 4,
                "width": 18,
              },
              undefined,
            ]
          }
        />
      </View>
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        blurOnSubmit={true}
        editable={false}
        keyboard="default"
        keyboardType="default"
        multiline={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder="Look for ..."
        placeholderTextColor="#707070"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
              "color": "#4a4a4a",
              "flex": 1,
              "fontFamily": undefined,
              "fontSize": 16,
              "fontWeight": "400",
              "minHeight": 42,
              "paddingHorizontal": 4,
              "paddingVertical": 7.5,
              "width": "100%",
            },
            Object {
              "color": "#707070",
            },
            undefined,
            undefined,
            undefined,
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "backgroundColor": "#ffffff",
        "borderRadius": 3,
        "paddingHorizontal": 12,
        "position": "absolute",
        "top": 44,
        "width": "100%",
      }
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;

exports[`<InputSearcher> when {hint} 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "marginBottom": 20,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "#cdd2d4",
            "borderRadius": 2.5,
            "borderWidth": 1,
            "display": "flex",
            "flexDirection": "row",
            "paddingHorizontal": 8,
          },
          false,
          false,
          false,
          undefined,
        ]
      }
    >
      <View
        pointerEvents="none"
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingHorizontal": 4,
          }
        }
      >
        <Image
          resizeMode="contain"
          source={
            Object {
              "testUri": "../../../components/Icon/assets/search.png",
            }
          }
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "height": 20,
                "width": 20,
              },
              Object {
                "alignSelf": "flex-start",
                "height": 18,
                "marginRight": 4,
                "width": 18,
              },
              undefined,
            ]
          }
        />
      </View>
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        blurOnSubmit={true}
        editable={true}
        keyboard="default"
        keyboardType="default"
        multiline={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder="Look for ..."
        placeholderTextColor="#707070"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
              "color": "#4a4a4a",
              "flex": 1,
              "fontFamily": undefined,
              "fontSize": 16,
              "fontWeight": "400",
              "minHeight": 42,
              "paddingHorizontal": 4,
              "paddingVertical": 7.5,
              "width": "100%",
            },
            false,
            undefined,
            undefined,
            undefined,
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={2}
      style={
        Array [
          Object {
            "color": "#4a4a4a",
            "fontFamily": undefined,
            "fontWeight": "400",
            "textAlignVertical": "center",
          },
          Object {
            "fontSize": 12,
            "lineHeight": 16,
          },
          Object {
            "color": "#707070",
          },
          Object {
            "marginTop": 4,
            "maxWidth": "100%",
            "paddingLeft": 0,
          },
        ]
      }
    >
      Hello World
    </Text>
  </View>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "backgroundColor": "#ffffff",
        "borderRadius": 3,
        "paddingHorizontal": 12,
        "position": "absolute",
        "top": 44,
        "width": "100%",
      }
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;

exports[`<InputSearcher> when {label} 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "marginBottom": 20,
        },
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={2}
      style={
        Array [
          Object {
            "color": "#4a4a4a",
            "fontFamily": undefined,
            "fontWeight": "400",
            "textAlignVertical": "center",
          },
          Object {
            "fontFamily": undefined,
            "fontSize": 14,
            "fontWeight": "600",
            "lineHeight": 16,
          },
          Object {
            "color": "#707070",
          },
          Object {
            "color": "#4a4a4a",
            "marginBottom": 8,
          },
        ]
      }
    >
      Hello World
    </Text>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "#cdd2d4",
            "borderRadius": 2.5,
            "borderWidth": 1,
            "display": "flex",
            "flexDirection": "row",
            "paddingHorizontal": 8,
          },
          false,
          false,
          false,
          undefined,
        ]
      }
    >
      <View
        pointerEvents="none"
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingHorizontal": 4,
          }
        }
      >
        <Image
          resizeMode="contain"
          source={
            Object {
              "testUri": "../../../components/Icon/assets/search.png",
            }
          }
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "height": 20,
                "width": 20,
              },
              Object {
                "alignSelf": "flex-start",
                "height": 18,
                "marginRight": 4,
                "width": 18,
              },
              undefined,
            ]
          }
        />
      </View>
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        blurOnSubmit={true}
        editable={true}
        keyboard="default"
        keyboardType="default"
        multiline={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder="Look for ..."
        placeholderTextColor="#707070"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
              "color": "#4a4a4a",
              "flex": 1,
              "fontFamily": undefined,
              "fontSize": 16,
              "fontWeight": "400",
              "minHeight": 42,
              "paddingHorizontal": 4,
              "paddingVertical": 7.5,
              "width": "100%",
            },
            false,
            undefined,
            undefined,
            undefined,
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "backgroundColor": "#ffffff",
        "borderRadius": 3,
        "paddingHorizontal": 12,
        "position": "absolute",
        "top": 44,
        "width": "100%",
      }
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;

exports[`<InputSearcher> when {onChange} 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "marginBottom": 20,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "#cdd2d4",
            "borderRadius": 2.5,
            "borderWidth": 1,
            "display": "flex",
            "flexDirection": "row",
            "paddingHorizontal": 8,
          },
          false,
          false,
          false,
          undefined,
        ]
      }
    >
      <View
        pointerEvents="none"
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingHorizontal": 4,
          }
        }
      >
        <Image
          resizeMode="contain"
          source={
            Object {
              "testUri": "../../../components/Icon/assets/search.png",
            }
          }
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "height": 20,
                "width": 20,
              },
              Object {
                "alignSelf": "flex-start",
                "height": 18,
                "marginRight": 4,
                "width": 18,
              },
              undefined,
            ]
          }
        />
      </View>
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        blurOnSubmit={true}
        editable={true}
        keyboard="default"
        keyboardType="default"
        multiline={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder="Look for ..."
        placeholderTextColor="#707070"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
              "color": "#4a4a4a",
              "flex": 1,
              "fontFamily": undefined,
              "fontSize": 16,
              "fontWeight": "400",
              "minHeight": 42,
              "paddingHorizontal": 4,
              "paddingVertical": 7.5,
              "width": "100%",
            },
            false,
            undefined,
            undefined,
            undefined,
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "backgroundColor": "#ffffff",
        "borderRadius": 3,
        "paddingHorizontal": 12,
        "position": "absolute",
        "top": 44,
        "width": "100%",
      }
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;

exports[`<InputSearcher> when {style} 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "marginBottom": 20,
        },
        Object {
          "backgroundColor": "rgba(0,255,0,0.25)",
          "padding": 10,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "#cdd2d4",
            "borderRadius": 2.5,
            "borderWidth": 1,
            "display": "flex",
            "flexDirection": "row",
            "paddingHorizontal": 8,
          },
          false,
          false,
          false,
          undefined,
        ]
      }
    >
      <View
        pointerEvents="none"
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingHorizontal": 4,
          }
        }
      >
        <Image
          resizeMode="contain"
          source={
            Object {
              "testUri": "../../../components/Icon/assets/search.png",
            }
          }
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "height": 20,
                "width": 20,
              },
              Object {
                "alignSelf": "flex-start",
                "height": 18,
                "marginRight": 4,
                "width": 18,
              },
              undefined,
            ]
          }
        />
      </View>
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        blurOnSubmit={true}
        editable={true}
        keyboard="default"
        keyboardType="default"
        multiline={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder="Look for ..."
        placeholderTextColor="#707070"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
              "color": "#4a4a4a",
              "flex": 1,
              "fontFamily": undefined,
              "fontSize": 16,
              "fontWeight": "400",
              "minHeight": 42,
              "paddingHorizontal": 4,
              "paddingVertical": 7.5,
              "width": "100%",
            },
            false,
            undefined,
            undefined,
            undefined,
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "backgroundColor": "#ffffff",
        "borderRadius": 3,
        "paddingHorizontal": 12,
        "position": "absolute",
        "top": 44,
        "width": "100%",
      }
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;

exports[`<InputSearcher> when {value} 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "marginBottom": 20,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "#cdd2d4",
            "borderRadius": 2.5,
            "borderWidth": 1,
            "display": "flex",
            "flexDirection": "row",
            "paddingHorizontal": 8,
          },
          false,
          false,
          false,
          undefined,
        ]
      }
    >
      <View
        pointerEvents="none"
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingHorizontal": 4,
          }
        }
      >
        <Image
          resizeMode="contain"
          source={
            Object {
              "testUri": "../../../components/Icon/assets/search.png",
            }
          }
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "height": 20,
                "width": 20,
              },
              Object {
                "alignSelf": "flex-start",
                "height": 18,
                "marginRight": 4,
                "width": 18,
              },
              undefined,
            ]
          }
        />
      </View>
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        blurOnSubmit={true}
        editable={true}
        keyboard="default"
        keyboardType="default"
        multiline={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder="Look for ..."
        placeholderTextColor="#707070"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
              "color": "#4a4a4a",
              "flex": 1,
              "fontFamily": undefined,
              "fontSize": 16,
              "fontWeight": "400",
              "minHeight": 42,
              "paddingHorizontal": 4,
              "paddingVertical": 7.5,
              "width": "100%",
            },
            false,
            undefined,
            undefined,
            undefined,
          ]
        }
        underlineColorAndroid="transparent"
        value="Madrid"
      />
    </View>
  </View>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "backgroundColor": "#ffffff",
        "borderRadius": 3,
        "paddingHorizontal": 12,
        "position": "absolute",
        "top": 44,
        "width": "100%",
      }
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;
